File: src/Parrot.Renderers/ConditionalRenderer.cs
namespace Parrot.Renderers
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using Parrot.Infrastructure;
    using Parrot.Nodes;
    using Parrot.Renderers.Infrastructure;

    public class ConditionalRenderer : HtmlRenderer
    {
        public ConditionalRenderer(IHost host) : base(host)
        {
        }

        public override IEnumerable<string> Elements
        {
            get { yield return "conditional"; }
        }

        public override void Render(IParrotWriter writer, IRendererFactory rendererFactory, Statement statement, IDictionary<string, object> documentHost, object model)
        {
            var localModel = GetLocalModel(documentHost, statement, model);

            string statementToOutput = "default";

            if (localModel != null)
            {
                statementToOutput = localModel.ToString();
            }

            //check that there is a parameter

            foreach (var child in statement.Children)
            {
                string value;
                var valueWriter = Host.CreateWriter();

                //get string value
                var renderer = rendererFactory.GetRenderer(child.Name);
                if (renderer is StringLiteralRenderer)
                {
                    renderer.Render(valueWriter, rendererFactory, child, documentHost, model);
                    value = valueWriter.Result();
                }
                else
                {
                    value = child.Name;
                }

                if (value.Equals(statementToOutput, StringComparison.OrdinalIgnoreCase))
                {
                    //render only the child
                    RenderChildren(writer, child, rendererFactory, documentHost, model);
                    return;
                }
            }

            var defaultChild = statement.Children.SingleOrDefault(s => s.Name.Equals("default", StringComparison.OrdinalIgnoreCase));
            if (defaultChild != null)
            {
                RenderChildren(writer, defaultChild, rendererFactory, documentHost, model);
            }
        }
    }
}

File: src/Parrot.Javascript/scripts/Renderers/conditionalRenderer.ts
///<reference path="../Parser/parser.ts" />
///<reference path="./irenderer.ts" />
///<reference path="./rendererProvider.ts" />
///<reference path="../Infrastructure/ObjectModelValueProvider.ts" />
///<reference path="../Infrastructure/ValueTypeProvider.ts" />
///<reference path="../exceptions.ts" />
///<reference path="./tagBuilder.ts" />
///<reference path="./baseRenderer.ts" />
///<reference path="./selfClosingRenderer.ts" />
///<reference path="./stringRenderer.ts" />

class ConditionalRenderer extends HtmlRenderer {
    defaultChildTag: string = "div";
    rendererProvider: RendererProvider;
    elements: string[] = ["conditional"];

    render(statement: Statement, host: any[], model: any, rendererProvider: RendererProvider): string {
        this.rendererProvider = rendererProvider;
        //get the local model
        //var localModel = this.getLocalModelValue(host, statement, model);
        var localModel = this.getLocalModelValue(host, statement, model);

        var statementToOutput = "default";

        if (localModel != null) {
            statementToOutput = localModel.toString();
        }

        for (var i in statement.children) {
            var child = statement.children[i];
            var value = "";

            var renderer = rendererProvider.getRenderer(child.name);
            if (renderer instanceof StringRenderer) {
                value = renderer.render(child, host, model, rendererProvider);
            } else {
                value = child.name;
            }

            if (value == statementToOutput) {
                return this.renderChildren(child, host, model, this.defaultChildTag);
            }

            for (var i in statement.children) {
                var child = statement.children[i];
                if (child.name == "default") {
                    return this.renderChildren(child, host, model, this.defaultChildTag);
                }
            }
        }
    }
}